packages/hash/shared/src/save.ts

```diff
@@ -201,7 +202,10 @@ const insertBlocks = defineOperation(
             componentId: componentNodeToId(node),
             accountId,
             // @todo support new non-text nodes
+
             entity: {
+              // This causes an issue with variant blocks like the embed block
+              // where a particular variant gets reset to the default variant.
               entityProperties: node.type.isTextblock
                 ? textBlockNodeToEntityProperties(node)
                 : {},
```

Comment:
This is responsible for the reset issue that happens with the youtube block. When we insert a youtube block, it has a draft child entity that has properties
```
{ embedType: "Youtube" }
```
Since we don't pass the entity properties too, it get's reset to the default embed block.
This also prevents loading an entity into a newly added/draft block.

Reply:
Ah, looks like we had a todo for this already

```
// @todo support new non-text nodes
```

Thanks for discovering this as the cause